operator-common/src/main/java/io/strimzi/operator/common/operator/resource/DeploymentOperator.java

```diff
@@ -61,25 +59,28 @@ protected Integer currentScale(String namespace, String name) {
      * @param namespace The namespace of the deployment
      * @param name The name of the deployment
      * @param operationTimeoutMs The timeout
+     * @param reasons - why the pod needs to be terminated
      * @return A future which completes when all the pods in the deployment have been restarted.
      */
-    public Future<Void> rollingUpdate(Reconciliation reconciliation, String namespace, String name, long operationTimeoutMs) {
+    public Future<Void> rollingUpdate(Reconciliation reconciliation, String namespace, String name, long operationTimeoutMs, RestartReasons reasons) {
         return getAsync(namespace, name)
-                .compose(deployment -> deletePod(reconciliation, namespace, name))
+                .compose(deployment -> restartSingletonPod(reconciliation, namespace, name, reasons, operationTimeoutMs))
                 .compose(ignored -> readiness(reconciliation, namespace, name, 1_000, operationTimeoutMs));
     }
 
     /**
-     * Asynchronously delete the given pod.
+     * Asynchronously restart the given pod.
      * @param reconciliation The reconciliation
      * @param namespace The namespace of the pod.
      * @param name The name of the pod.
+     * @param reasons why the pod needs to be terminated
+     * @param timeoutMs how long the restart operation has to complete
      * @return A Future which will complete once all the pods has been deleted.
      */
-    public Future<ReconcileResult<Pod>> deletePod(Reconciliation reconciliation, String namespace, String name) {
+    public Future<Void> restartSingletonPod(Reconciliation reconciliation, String namespace, String name, RestartReasons reasons, long timeoutMs) {
```

Comment:
This is wrong. Delete and restart pod are different things. Deployment also doesn't give you control over restarts the way it is used in Strimzi. So you should not pull this into it.

Reply:
I renamed it to match the semantics of [where it's being called from](https://github.com/strimzi/strimzi-kafka-operator/blob/72a0260e828adad35d44e590fdbebd4b407df275/operator-common/src/main/java/io/strimzi/operator/common/operator/resource/DeploymentOperator.java#L66). So either renaming it to restart vs. delete is wrong, or calling it from `rollingUpdate` with a `Reconciliation` attached is wrong.

Reply:
It deletes the pod. The StatefulSet controller is what starts it again. Especially since we are removing the StatefulSets and in the future something else will start it it is important to call it what it is.

Reply:
Okay, that's fine (although `KafkaRoller` has similar naming issues), this'll be reverted when I reduce scope back to the `KafkaRoller` only.